clients/cli/src/prover.rs

```diff
@@ -261,10 +276,26 @@ async fn main() {
                 }),
             );
             progress_time = SystemTime::now();
-            client
-                .send(Message::Binary(progress.encode_to_vec()))
-                .await
-                .unwrap();
+
+            let mut retries = 0;
+            let max_retries = 5;
+            loop {
+                if let Err(e) = client.send(Message::Binary(progress.encode_to_vec())).await {
+                    eprintln!("Failed to send message: {:?}, attempt {}/{}", e, retries + 1, max_retries);
+        
+                    retries += 1;
+                    if retries >= max_retries {
+                        eprintln!("Max retries reached, exiting...");
+                        break;
+                    }
+        
+                    // Optionally, add a delay before retrying
+                    tokio::time::sleep(tokio::time::Duration::from_secs(2)).await;
```

Comment:
Can we do exponential backoff here? (Multiply sleep delay by 2 each time.)

Reply:
okay, let me update

Reply:
-                    // Optionally, add a delay before retrying
-                    tokio::time::sleep(tokio::time::Duration::from_secs(2)).await;
+                    tokio::time::sleep(tokio::time::Duration::from_secs(u64::pow(2, retries))).await;

Reply:
I updated, please help me check and merge.

Thank you